### PR TITLE
improvement(runner): associate a public IP from a pool

### PR DESCRIPTION
**Pull request by enaydanov**
The IP pool is a list of pre-allocated IP address tagged with 'sct-runner-pool' tag.
The use of SCT can provide a name of a pool to use (the value of such tag.)

This PR implements such functionality for AWS and GCP. Azure implementation is missed (and not required now.)

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [x] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [x] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- [x] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: sdcm/utils/gce_utils.py [hunk 1]
```diff
@@ -40,6 +40,7 @@
     # us-east1 zones: b, c, and d. Details: https://cloud.google.com/compute/docs/regions-zones#locations
     # Currently choose only zones c and d as zone b frequently fails allocating resources.
     'us-east1': 'cd',
+    'us-east4': 'abc',
```
Comment (fruch) [inline sdcm/utils/gce_utils.py:43]: Can we do this in a separate commit ?

File: requirements.in [hunk 2]
```diff
@@ -49,6 +49,6 @@ azure-mgmt-resourcegraph==8.0.0
 pydantic==1.10.8
 azure-storage-blob==12.16.0
 questionary==1.10.0
-pbr==1.4
```
Comment (fruch) [inline requirements.in:52]: Why did we need to upgrade this one ?

**Comment by fruch** [comment]
@enaydanov where is the code that would create those pools ? in siren-tests ?

File: requirements.in (hunk 2 above)
Comment (enaydanov) [reply requirements.in:52]: Can't build a docker image without it :(

**Comment by enaydanov** [comment]
The pools created manually, this process requires to interact with devops to add IPs to SG they maintain.

File: sdcm/utils/gce_utils.py (hunk 1 above)
Comment (enaydanov) [reply sdcm/utils/gce_utils.py:43]: Sure.

**Comment by fruch** [comment]
> The pools created manually, this process requires to interact with devops to add IPs to SG they maintain.
> 
> And yes, it's for siren-tests.

I would recommend that it would at least be documented in the correct repo, with copy-pastable instructions as much as possible.

File: sdcm/sct_runner.py [hunk 3]
```diff
@@ -539,6 +550,22 @@ def _create_instance(self,
         LOGGER.info("Instance created. Waiting until it becomes running... ")
         instance.wait_until_running()
 
+        if address_pool:
+            LOGGER.info("Associating an EIP from `%s' pool...", address_pool)
+            unassigned_addresses = [
```
Comment (soyacz) [inline sdcm/sct_runner.py:555]: maybe use `available_addresses`?

File: sdcm/sct_runner.py [hunk 4]
```diff
@@ -733,20 +760,38 @@ def _create_instance(self,
...
         disks = [disk_from_image(disk_type=f"projects/{self.project_name}/zones/{region_az}/diskTypes/pd-ssd",
                                  disk_size_gb=root_disk_size_gb or self.instance_root_disk_size(test_duration),
                                  boot=True,
                                  source_image=base_image,
                                  auto_delete=True)]
 
+        if address_pool:
+            LOGGER.info("Use External IP address from `%s' pool...", address_pool)
+            addresses_client, info = get_gce_compute_addresses_client()
+            unassigned_addresses = list(
```
Comment (soyacz) [inline sdcm/sct_runner.py:775]: `available_addresses`?

**Review by soyacz — COMMENTED** [review]
Can you explain why we need this pool at all? What problem it solves?

File: requirements.in (hunk 2 above)
Comment (fruch) [reply requirements.in:52]: let's remove it from requriements.in, `hdrhistogram` is using it, not SCT directly.

Comment (enaydanov) [reply requirements.in:52]: Can't rebuild a Docker image right now without updating PyYAML to 6.0.1 and awscli to >=1.29.4 because of the problem with fresh Cython.

Comment (fruch) [reply requirements.in:52]: Why in the first place we needed upgrade of Google tools ? What wasn't working ?

**Review by soyacz — APPROVED** [review]
LGTM

File: requirements.in (hunk 2 above)
Comment (fruch) [reply requirements.in:52]: Why in the first place we needed upgrade of Google tools ? What wasn't working ?

**Review by rayakurl — APPROVED** [review]
approved

File: requirements.in (hunk 2 above)
Comment (fruch) [reply requirements.in:52]: o.k. next one that would try to build a new image, is heads for a surprise
can you open an issue with the details of what you do know, and which version are currently broken ?

Comment (enaydanov) [reply requirements.in:52]: Sure, I'll do.

Comment (rayakurl) [reply requirements.in:52]: @enaydanov please add a link to the issue here. thanks

Comment (enaydanov) [reply requirements.in:52]: https://github.com/scylladb/scylla-cluster-tests/issues/6389
